packages/core/src/codecs/octetstream-codec.ts

```diff
@@ -140,98 +166,132 @@ export default class OctetstreamCodec implements ContentCodec {
                     : bytes.readUInt32LE(0);
 
             default: {
-                let result = 0;
-                let negative;
-
-                if (bigendian) {
-                    result = bytes.reduce((prev, curr) => prev << (8 + curr));
-                    negative = bytes.readInt8(0) < 0;
-                } else {
-                    result = bytes.reduceRight((prev, curr) => prev << (8 + curr));
-                    negative = bytes.readInt8(dataLength - 1) < 0;
-                }
-
-                if (signed && negative) {
-                    result -= 1 << (8 * dataLength);
-                }
-
+                const result = bigEndian
+                    ? signed
+                        ? bytes.readIntBE(0, dataLength / 8)
+                        : bytes.readUIntBE(0, dataLength / 8)
+                    : signed
+                    ? bytes.readIntLE(0, dataLength / 8)
+                    : bytes.readUIntLE(0, dataLength / 8);
                 // warn about numbers being too big to be represented as safe integers
                 if (!Number.isSafeInteger(result)) {
                     warn("Result is not a safe integer");
                 }
-
                 return result;
             }
         }
     }
 
-    private numberToValue(bytes: Buffer, options: { dataLength: number; bigendian: boolean }): number {
-        const { dataLength, bigendian } = options;
+    private numberToValue(bytes: Buffer, options: { dataLength: number; bigEndian: boolean }): number {
+        const { dataLength, bigEndian } = options;
         switch (dataLength) {
-            case 2:
-                return getFloat16(new DataView(bytes.buffer), bytes.byteOffset, !bigendian);
-            case 4:
-                return bigendian ? bytes.readFloatBE(0) : bytes.readFloatLE(0);
+            case 16:
+                return getFloat16(new DataView(bytes.buffer), bytes.byteOffset, !bigEndian);
+            case 32:
+                return bigEndian ? bytes.readFloatBE(0) : bytes.readFloatLE(0);
 
-            case 8:
-                return bigendian ? bytes.readDoubleBE(0) : bytes.readDoubleLE(0);
+            case 64:
+                return bigEndian ? bytes.readDoubleBE(0) : bytes.readDoubleLE(0);
 
             default:
-                throw new Error("Wrong buffer length for type 'number', must be 2, 4, 8, or is " + dataLength);
+                throw new Error("Wrong buffer length for type 'number', must be 16, 32, or 64 is " + dataLength);
         }
     }
 
+    private objectToValue(bytes: Buffer, schema?: DataSchema): DataSchemaValue {
+        if (schema?.type !== "object") {
+            throw new Error("Schema must be of type 'object'");
+        }
+
+        const result: { [key: string]: unknown } = {};
+        for (const propertyName in schema.properties) {
+            const propertySchema = schema.properties[propertyName];
+            result[propertyName] = this.bytesToValue(bytes, propertySchema);
```

Comment:
Now I think I understand why you moved the parameters to the schema. I still keep then as global parameters in the content-type and then apply them to all the subschema. It is very unlikely that the same platform will encode the properties in a different format. If it is BE it is BE for all the payloads.

Reply:
Yes please keep everything in the content-type for now.

Reply:
I'd use the length parameter defined in the content type as an array length. If in the schema we are trying to access outside its ranges (either by specifying a bitOffset or a bitLenght) then it is a parsing error. If we want to downgrade gracefully we can take whatever is possible and log a warning.

Reply:
Perfect, I think that's taken care of by my most recent commit (28b135c346b922a6a22c7c34c8fd9324dda353a4) then